docroot/modules/custom/layout_builder_custom/src/Plugin/Display/ListBlock.php

```diff
@@ -333,6 +356,11 @@ public function blockForm(ViewsBlock $block, array &$form, FormStateInterface $f
         $more_link_help_text = $this->t('Start typing to see a list of results. Click to select.');
       }
 
+      $more_link_text = $this->getOption('more_link_text');
+      if (empty($more_link_text)) {
+        $more_link_text = $this->t('Enter your custom text for the button');
```

Comment:
This seems more like the text you would want in `#description` than to be set to this value.